api/v2/rpc.go

```diff
@@ -164,93 +193,23 @@ func (s *NotificationServer) DeleteNotification(ctx context.Context, req *pb.Del
 		return nil, status.Error(codes.InvalidArgument, "Failed to provide notification name")
 	}
 
-	err := s.Store.DeleteNotification(req.GetName())
+	tx, err := s.Store.Begin()
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
```

Comment:
any reason you don't defer the rollback here and instead explicitly call it for each branch below?